src/PrestaShopBundle/Twig/Layout/MenuBuilder.php

```diff
@@ -101,19 +101,40 @@ public function convertTabsToBreadcrumbLinks(Tab $currentTab, array $tabAncestor
 
     private function convertTabToMenuLink(Tab $tab): MenuLink
     {
-        if (!empty($tab->getRouteName())) {
-            $href = $this->urlGenerator->generate($tab->getRouteName());
-        } else {
-            $href = $this->context->getAdminLink($tab->getClassName());
-        }
-
         return new MenuLink(
             name: $this->getBreadcrumbLabel($tab),
-            href: $href,
+            href: $this->getLinkFromTab($tab),
             icon: 'icon-' . $tab->getClassName(),
         );
     }
 
+    /**
+     * @return array<int, MenuLink>
+     */
+    public function buildNavigationTabs(Tab $tab): array
+    {
+        $currentLevelTabs = $this->tabRepository->findByParentId($tab->getIdParent());
+        $navigationTabs = [];
+
+        /* @var $currentLevelTab Tab */
+        foreach ($currentLevelTabs as $currentLevelTab) {
+            $tabLang = $currentLevelTab->getTabLangByLanguageId($this->getContextLanguageId());
+            $menuLink = new MenuLink(
```

Comment:
This `MenuLink` DTO was created as a very simple one, maybe we'll have to improve it (add fields that are used everywhere) Or create alternative ones like a dedicated `TabLink` DTO or something similar (not in this PR as well)

Reply:
I agree, might as well create a new TabLink